app/src/main/java/org/mozilla/fenix/tabstray/SecureTabsTrayBinding.kt

```diff
@@ -36,8 +38,18 @@ class SecureTabsTrayBinding(
                 if (state.selectedPage == Page.PrivateTabs &&
                     !settings.allowScreenshotsInPrivateMode) {
                     fragment.secure()
+                    dialog.window.apply {
+                        this?.addFlags(
+                            WindowManager.LayoutParams.FLAG_SECURE
+                        )
+                    }
                 } else if (!settings.lastKnownMode.isPrivate) {
                     fragment.removeSecure()
+                    dialog.window.apply {
+                        this?.clearFlags(
+                            WindowManager.LayoutParams.FLAG_SECURE
+                        )
+                    }
```

Comment:
You could clear flags directly without needing to use `apply` or `this`. 

I found [this blog post](https://medium.com/@fatihcoskun/kotlin-scoping-functions-apply-vs-with-let-also-run-816e4efb75f5) useful for understanding what would be the best use cases for the various types of scoping functions which you might find interesting too. 🙂 

```suggestion
                    dialog.window?.addFlags(WindowManager.LayoutParams.FLAG_SECURE)
                } else if (!settings.lastKnownMode.isPrivate) {
                    fragment.removeSecure()
                    dialog.window?.clearFlags(WindowManager.LayoutParams.FLAG_SECURE)
```